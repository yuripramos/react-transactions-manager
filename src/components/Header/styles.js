import styled, { css } from "styled-components";
import { rem, media } from "../../styles/tools";
import { white, green } from "../../styles/settings";

export const HeaderWrapper = styled.header`
  height: ${rem(58)};
  padding-top: ${rem(10)};
  background: ${green};
  text-align: center;
  ${media.md(css`
    text-align: center;
  `)};
  ${media.lg(css`
    height: 7vh;
    padding-top: ${rem(6)};
  `)};
`;

export const Title = styled.h1`
  font-size: ${rem(25)};
  color: ${white};
  text-shadow: 0.1875rem 0.1rem rgba(132, 129, 129, 0.15);
`;

export const TitleWrapper = styled.div`
  display: inline-flex;
  width: 50%;
  font-size: ${rem(28)};
  ${media.xs(css`
    width: 95%;
    justify-content: flex-end;
  `)};
`;

export const SubTitle = styled.h2`
  font-size: ${rem(10)};
  text-align: center;
  color: ${white};
  text-shadow: ${rem(3)} ${rem(3)} rgba(0, 0, 0, 0.25);
  ${media.md(css`
    font-size: ${rem(24)};
    text-align: left;
  `)};
`;

export const InputWrapper = styled.div`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 100%;
`;

export const ListIcons = styled.div`
  display: flex;
  width: ${rem(200)};
  margin-top: ${rem(15)};
`;

export const FeaturedImage = styled.div`
  width: 100%;
  height: 100%;
  margin: 0;
`;

export const ImgHeader = styled.img`
  width: ${rem(60)};
  height: ${rem(60)};
  ${media.lg(css`
    width: ${rem(80)};
    height: ${rem(80)};
  `)};
`;
