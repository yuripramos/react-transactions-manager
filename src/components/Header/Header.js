import React, { Component } from "react";
import { shape, func } from "prop-types";
import { Container, Row, Column } from "../../styles/grid";
import SidebarMenu from "../common/SidebarMenu";
import Logo from "../../assets/currency.png";
import { HeaderWrapper, TitleWrapper } from "./styles";
import { slide as Menu } from "react-burger-menu";
import Hide from "../common/Hide";
import Icon from "../common/Icon";
import TransactionList from "../Content/TransactionList";
import { rem } from "../../styles/tools";
import { white, black } from "../../styles/settings";

const bmStyles = {
  bmBurgerButton: {
    display: "inline-block",
    position: "absolute",
    top: `${rem(20)}`,
    left: `${rem(25)}`,
    padding: `${rem(2)}`,
    cursor: "pointer"
  },
  bmMenuWrap: {
    left: 0
  }
};

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      menuOpen: false
    };

    this.onToggleMenu = this.onToggleMenu.bind(this);
    this.closeMenu = this.closeMenu.bind(this);
  }
  onToggleMenu({ isOpen }) {
    this.setState({ menuOpen: isOpen });
  }

  closeMenu() {
    this.setState({ menuOpen: false });
  }

  render() {
    const {
      shipsArray,
      ShowDetails,
      detailedShip,
      isFilled,
      history
    } = this.props;
    const { menuOpen } = this.state;
    return (
      <HeaderWrapper>
        <Container onClick={this.closeMenu}>
          <Hide below="md">
            <Row>
              <Column>
                <SidebarMenu currentLocation={history.location.pathname} />
              </Column>
              <Column>
                <TitleWrapper>
                  <img
                    src={Logo}
                    alt="logo financial"
                    width="44px"
                    height="44px"
                  />
                </TitleWrapper>
              </Column>
            </Row>
          </Hide>
          <Hide above="md">
            {isFilled && (
              <Menu
                isOpen={menuOpen}
                onStateChange={this.onToggleMenu}
                customBurgerIcon={
                  menuOpen === true ? (
                    <Icon name="MenuClose" color={white} />
                  ) : (
                    <Icon name="Menu" color={black} />
                  )
                }
                width="70%"
                styles={bmStyles}
              >
                <TransactionList
                  ships={shipsArray}
                  ShowDetails={ShowDetails}
                  detailedShip={detailedShip}
                />
              </Menu>
            )}
            <TitleWrapper>
              <img src={Logo} alt="logo loadsmart" width="50px" height="50px" />
            </TitleWrapper>
          </Hide>
        </Container>
      </HeaderWrapper>
    );
  }
}

export default Header;

Header.propTypes = {
  history: shape({
    push: func
  })
};
