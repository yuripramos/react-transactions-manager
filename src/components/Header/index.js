import Header from "./Header";
import { connect } from "redux-zero/react";
import { withRouter } from "react-router-dom";

import transactionsActions from "../../actions/Transactions";

export default connect(
  ({ isFilled, shipsArray, ShowDetails, detailedShip }) => ({
    isFilled,
    shipsArray,
    ShowDetails,
    detailedShip
  }),
  transactionsActions
)(withRouter(Header));
