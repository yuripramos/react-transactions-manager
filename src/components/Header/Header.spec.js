import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

let props;

beforeEach(() => {
  props = {
    history: {
      push: jest.fn(),
      location: {
        pathname: ""
      }
    }
  };
});

describe("Header component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<Header {...props}/>)).toMatchSnapshot();
  });

});
