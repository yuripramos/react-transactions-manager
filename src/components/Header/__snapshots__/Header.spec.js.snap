// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot with props 1`] = `
<styled.header>
  <Container>
    <Row>
      <Column>
        <styled.div>
          <IconGenerator
            color="#333"
            height="56px"
            name="Logo"
            width="490px"
          />
        </styled.div>
      </Column>
    </Row>
    <Row>
      <Column>
        <SidebarMenu
          currentLocation=""
        />
      </Column>
    </Row>
  </Container>
</styled.header>
`;
